packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -940,28 +960,17 @@ mixin _RenderTheaterMixin on RenderBox {
     }
   }
 
-  @override
-  bool get sizedByParent => true;
-
-  @override
-  void performLayout() {
-    final Iterator<RenderBox> iterator = _childrenInPaintOrder().iterator;
-    // Same BoxConstraints as used by RenderStack for StackFit.expand.
-    final BoxConstraints nonPositionedChildConstraints = BoxConstraints.tight(constraints.biggest);
+  void layoutChild(RenderBox child, BoxConstraints nonPositionedChildConstraints) {
+    final StackParentData childParentData = child.parentData! as StackParentData;
```

Comment:
Has it been asserted elsewhere that `child.parentData` vends `StackParentData`? If not, consider asserting with a descriptive error message prior to unboxing.

Reply:
It's guaranteed to be StackParentData here and using it this way is the established pattern in the framework.